nativelib/src/main/resources/dirent.c

```diff
@@ -1,19 +1,29 @@
+#ifndef _WIN32
 #include <dirent.h>
+#else
+#include "os_win_dirent.h"
```

Comment:
Btw, Sockets are supported in this PR.
Also, Microsoft introduced Universal CRT SDK, which is partly used here as well, which mimics POSIX syntax. It was developed to support unix on windows. It's not going to support all features of course (it will not support pthreads for instance).

But longer I look into this port, I become more agree with you on rewriting javalib.
I started this work already as part of my third-party library for scala-native , but it will take a few months for me until I have first results.